lib/std/crypto/modes.zig

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2020 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+// Based on Go stdlib implementation
+
+const std = @import("../std.zig");
+const mem = std.mem;
+const debug = std.debug;
+
+/// Counter mode.
+pub fn ctr(comptime BlockCipher: anytype, block_cipher: BlockCipher, dst: []u8, src: []const u8, iv: [BlockCipher.block_size]u8, bigEndian: comptime bool) void {
+    debug.assert(dst.len >= src.len);
+    const block_size = BlockCipher.block_size;
+    var counter: [BlockCipher.block_size]u8 = undefined;
+    var counterInt: u128 = if (bigEndian)
```

Comment:
If `bigEndian` is replaced with the use of the `builtin.Endian` type you can use `mem.readInt` with the extra parameter.
But if you're tied to the target endianness then `readIntNative` is the way to go.

Reply:
This is not meant to be the platform endianness.

AES-GCM for example, encodes its counter as big endian, whereas AES-GCM-SIV encodes its counter as little endian.
This is how the constructions are defined, no matter what the platform is.

So, in a generic counter mode implementation, this has to be provided as a parameter.

Reply:
Oh, I see, you mean changing the type to `builtin.Endian`. Got it.

Reply:
Done, thanks, @LemonBoy !